datafusion/datasource-parquet/src/file_format.rs

```diff
@@ -1033,24 +1038,62 @@ impl MetadataFetch for ObjectStoreFetch<'_> {
 /// through [`ParquetFileReaderFactory`].
 ///
 /// [`ParquetFileReaderFactory`]: crate::ParquetFileReaderFactory
-pub async fn fetch_parquet_metadata(
-    store: &dyn ObjectStore,
-    meta: &ObjectMeta,
+pub async fn fetch_parquet_metadata<F: MetadataFetch>(
```

Comment:
I wonder now if this function should go to a separate module (a `utils.rs` or similar?). This is because the `file_format` refers to `reader` and `reader` refers to `file_format`. cc @alamb